helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -92,19 +98,42 @@ public ZkBaseDataAccessor(HelixZkClient zkClient) {
     _zkClient = zkClient;
   }
 
+  private synchronized HelixZkClient getNonZNRecordClient() {
+    if (_nonZNRecordClient == null) {
+      _nonZNRecordClient = new ZkClient.Builder().setConnection(
```

Comment:
Let's use make it clear in the variable name that this is a ZK client.

Reply:
done